crates/forge-runner/src/test_case_summary.rs

```diff
@@ -264,23 +266,77 @@ impl TestCaseSummary<Single> {
     }
 }
 
-/// Helper function to build `readable_text` from a run data.
-fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
-    let mut readable_text = String::new();
+/// Helper function to build readable text from a run data.
+fn build_readable_text(data: &[Felt252]) -> Option<String> {
+    let mut data_iter = data.iter().cloned();
+    let mut items = Vec::new();
 
-    for felt in data {
-        readable_text.push_str(&format!("\n    original value: [{felt}]"));
-        if let Some(short_string) = as_cairo_short_string(felt) {
-            readable_text.push_str(&format!(", converted to a string: [{short_string}]"));
-        }
+    while let Some(item) = format_next_felt(&mut data_iter) {
+        items.push(format!("\n{item}"));
     }
 
-    if readable_text.is_empty() {
+    if items.is_empty() {
         None
     } else {
-        readable_text.push('\n');
-        Some(readable_text)
+        let mut result = items.join("").to_string().replace('\n', "\n    ");
+        result.push('\n');
+        Some(result)
+    }
+}
+
+fn format_next_felt<T>(values: &mut T) -> Option<String>
+where
+    T: Iterator<Item = Felt252> + Clone,
+{
+    let first_felt = values.next()?;
+
+    if first_felt == felt_str!(BYTE_ARRAY_MAGIC, 16) {
+        if let Some(string) = try_format_string(values) {
+            return Some(string);
+        }
+    }
+
+    if let Some(short_string) = as_cairo_short_string(&first_felt) {
+        return Some(format!(
+            "original value: [{first_felt}], converted to a string: [{short_string}]"
+        ));
```

Comment:
Can we revamp this one too? Rn only strings are printed nicely, shortstrings could be too. Example from cairo-test:
![image](https://github.com/foundry-rs/starknet-foundry/assets/56825108/67998a51-d81b-4bcd-91d9-f868a4840676)

Reply:
Also i think a panic array output should be output as a whole instead of `n` formatted lines like that